### PR TITLE
fix(TDI-44750): add null fields in record's schema with json

### PR DESCRIPTION
**Pull request by undx**
https://jira.talendforge.org/browse/TDI-44750

When returning a Record from a Json file with null values, the null values are skipped from schema. 
So, the record's schema is not fully feeded.

### CODE REVIEW

**Review by rlecomte-talend — APPROVED** [review]
LGTM

File: azure-dls-gen2/src/main/java/org/talend/components/adlsgen2/common/format/json/JsonConverter.java [hunk 1]
```diff
@@ -68,10 +66,7 @@ public Schema inferSchema(final JsonObject record) {
 
     @Override
     public Record toRecord(final JsonObject record) {
-        if (schema == null || record.keySet().size() > schema.getEntries().size()) {
-            schema = inferSchema(record);
-        }
-        return convertJsonObjectToRecord(schema, record);
+        return convertJsonObjectToRecord(inferSchema(record), record);
```
Comment (clesaec) [inline azure-dls-gen2/src/main/java/org/talend/components/adlsgen2/common/format/json/JsonConverter.java:69]: rename convertJsonObjectToRecord in toRecord and make it public, it should be used.

File: azure-dls-gen2/src/main/java/org/talend/components/adlsgen2/common/format/json/JsonConverter.java [hunk 2]
```diff
@@ -34,8 +36,6 @@
 
 import lombok.extern.slf4j.Slf4j;
 
-import static java.util.stream.Collectors.toList;
-
 @Slf4j
 public class JsonConverter implements RecordConverter<JsonObject>, Serializable {
```
[the remark this answers is not in the document]
Comment (undx) [reply azure-dls-gen2/src/main/java/org/talend/components/adlsgen2/common/format/json/JsonConverter.java:40]: This is a fix issue, not a refactoring one... A work item should be created for that.

File: azure-dls-gen2/src/main/java/org/talend/components/adlsgen2/common/format/json/JsonConverter.java (hunk 1 above)
Comment (undx) [reply azure-dls-gen2/src/main/java/org/talend/components/adlsgen2/common/format/json/JsonConverter.java:69]: don't get the point here.

**Review by clesaec — APPROVED** [review]
:+1:

**Comment by undx** [comment]
obsolete

**Pull request closed without merging**